testing/rdiffbackupdeletetest.py

```diff
@@ -67,7 +66,13 @@ def assertNotFound(self, search):
 
     def tearDown(self):
         if os.path.exists(self.repo):
-            subprocess.check_call(['rm', '-Rf', self.repo])
+            for root, dirs, files in os.walk(self.repo, followlinks=False):
+                for d in dirs:
+                    os.chmod(os.path.join(root, d), 0o777)
+                for f in files:
+                    if not os.path.islink(os.path.join(root, f)):
+                        os.chmod(os.path.join(root, f), 0o777)
+            subprocess.check_call([b'rm', b'-Rf', self.repo])

```

Comment:
I know that we do it in the current (outdated) code, but we should try to avoid calling (OS specific) commands instead of using pure python. https://docs.python.org/3.8/library/os.html#os.walk contains an example on how to delete a complete tree, you would only need to adapt it slightly:

```python
for root, dirs, files in os.walk(top, topdown=False):
    for name in files:
        os.remove(os.path.join(root, name))
    for name in dirs:
        os.rmdir(os.path.join(root, name))
```

Extra points if you make it a generic function in commontest to reuse elsewhere...